### ISSUE TITLE
MacOS Issue: Symlink Creation and Permission Setting in Certificate Bundle Update

### ISSUE THREAD

**Issue by jwdsuhail** [BandarLabs/clickclickclick#2]
Hi,
I’m facing an issue on macOS where the script fails to create a symlink to the "certifi" certificate bundle or set its permissions. The relevant code is:

except FileNotFoundError:
    pass

print("Creating symlink to certifi certificate bundle")
os.symlink(relpath_to_certifi_cafile, openssl_cafile)

print("Setting permissions")
os.chmod(openssl_cafile, STAT_00775)

print("Update complete")

Could you provide guidance on debugging or fixing this issue on macOS? Are there any known workarounds for creating symlinks or setting permissions in this environment?

**Comment by mkagenius** [comment]
Did this happen during pip install <repo_tar.gz>?
I also use MacOS, so this is surprising.

(Btw you can directly use `main.py` to run the tool, the instructions are in 'As Script' section of Readme)

**Comment by jwdsuhail** [comment]
Yes, this issue occurred during the setup process, specifically when trying
to create a symlink to the certifi certificate bundle as described in the
code snippet.

Thank you for pointing out the option to use main.py directly! I'll try
that approach and update here if the issue persists

**Comment by mkagenius** [comment]
Can try this `pip install --upgrade certifi` to fix the certifi issue.

This seems more like an environment issue than code issue at the outset.
